src/dotnet/Telemetry.cs

```diff
@@ -146,5 +155,49 @@ private void TrackEventTask(string eventName, IDictionary<string, string> proper
                 return _commonProperties;
             }
         }
+        
+        private List<string> GetMacs()
+        {
+            NetworkInterface[] nics = NetworkInterface.GetAllNetworkInterfaces();
+            var macs = new List<string>();
+            
+            if (nics == null || nics.Length < 1)
+            {
+                macs.Add(string.Empty);
+                return macs;
+            }
+            
+            foreach (NetworkInterface adapter in nics)
+            {
+                IPInterfaceProperties properties = adapter.GetIPProperties();
+
+                PhysicalAddress address = adapter.GetPhysicalAddress();
+                byte[] bytes = address.GetAddressBytes();
+                macs.Add(string.Join("-", bytes.Select(x => x.ToString("X2"))));
```

Comment:
Also, looking below, we are just turning byte[]s into strings here. Then later, turning those strings back into byte[]s.  Why not just return List<byte[]> from here, and skip the back and forth?